### DISCUSSION TITLE
Endpoint with parameter that has forward "/" slashes

### DISCUSSION THREAD

**Discussion by NickDatLe** [fastapi/fastapi#7362] [Questions] [answered]
Is there a way to put a parameter that is part of the URL but it has forward-slashes?

For example:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/items/{path}")
def read_root(path:str="/inventory/armor/helmet):
    return path
```
With that route above, if I put example.com/items/this/is/my/path I get a 404.

**Comment by Kludex** [comment, marked as answer]
This may help: https://stackoverflow.com/questions/62770461/starlette-fastapi-route-path-components-with-forward-slash

**Comment by NickDatLe** [comment]
thanks for the reply; I've put the {path} as a query string parameter now.

```python
@app.get("/items")
def read_root(k:str =Query(None)):
    return path
```
This works and seems safe.

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex! :clap: :bow: 

Thanks for reporting back and closing the issue @NickDatLe  :+1:

**Comment by bryophyta** [comment]
If I've understood this issue correctly then there's now a solution in the [official docs](https://fastapi.tiangolo.com/tutorial/path-params/#path-convertor) using Starlette which is similar to one of the answers in the Stackoverflow.

**Reply by razzeee** [reply]
Does it matter if my thing is not a `path`, but like a regular name of something?

**Reply by ashleysommer** [reply]
No, it doesn't matter if it isn't really a path, using the `path` param type simply takes everything from that point on and converts it to a string parameter, including forward slashes. You can use it for any kind of identifier.

For any others searching this issue from google (like I was), the correct solution (hinted above) is to use a type param, with the `path` type, like this:

```python
@app.get("/v1/item/{identifier:path}")
async def get_item(identifier: str, request: Request):
    ...
```

Then a query to something like this: `http://localhost:5000/v1/item/2006/v3/6/large` will worth with the identifier being "2006/v3/6/large"
